File: airtime_api.py
import requests

API_KEY = 'YOUR_API_KEY'  # Replace with your actual API key
BASE_URL = 'https://api.yourairtimeservice.com'

headers = {
    'Authorization': f'Bearer {API_KEY}',
    'Content-Type': 'application/json'
}

# Check Balance
def check_balance():
    url = f'{BASE_URL}/api/v1/balance'
    response = requests.get(url, headers=headers)
    print(response.json())

# Purchase Airtime
def purchase_airtime(mobile_number, amount):
    url = f'{BASE_URL}/api/v1/topup'
    data = {
        'mobile_number': mobile_number,
        'amount': amount,
        'currency': 'USD'
    }
    response = requests.post(url, headers=headers, json=data)
    print(response.json())

# Check Transaction Status
def check_transaction_status(transaction_id):
    url = f'{BASE_URL}/api/v1/transaction/{transaction_id}'
    response = requests.get(url, headers=headers)
    print(response.json())

# Example usage
check_balance()
purchase_airtime('+1234567890', 10.00)
check_transaction_status('1234567890ABC')

File: airtime_api.js
const axios = require('axios');

const API_KEY = 'YOUR_API_KEY';
const BASE_URL = 'https://api.airtimeprovider.com';

const headers = {
    'Authorization': `Bearer ${API_KEY}`,
    'Content-Type': 'application/json'
};

// Check Balance
async function checkBalance() {
    try {
        const response = await axios.get(`${BASE_URL}/api/v1/balance`, { headers });
        console.log(response.data);
    } catch (error) {
        console.error(error);
    }
}

// Purchase Airtime
async function purchaseAirtime(mobile_number, amount) {
    try {
        const response = await axios.post(`${BASE_URL}/api/v1/topup`, {
            mobile_number,
            amount,
            currency: 'USD'
        }, { headers });
        console.log(response.data);
    } catch (error) {
        console.error(error);
    }
}

// Check Transaction Status
async function checkTransactionStatus(transaction_id) {
    try {
        const response = await axios.get(`${BASE_URL}/api/v1/transaction/${transaction_id}`, { headers });
        console.log(response.data);
    } catch (error) {
        console.error(error);
    }
}

// Example usage
checkBalance();
purchaseAirtime('+1234567890', 10.00);
checkTransactionStatus('1234567890ABC');
